lib/orderbook/OrderBook.ts

```diff
@@ -289,25 +289,37 @@ class OrderBook extends EventEmitter {
     // perform match. maker orders will be removed from the repository
     const tp = this.getTradingPair(order.pairId);
     const matchingResult = tp.match(order);
-
-    // instantiate the final response object
-    const result: PlaceOrderResult = {
-      internalMatches: [],
-      swapResults: [],
-      remainingOrder: matchingResult.remainingOrder,
+    /** Any portion of the placed order that could not be swapped or matched internally. */
+    let { remainingOrder } = matchingResult;
+    /** Local orders that matched with the placed order. */
+    const internalMatches: OwnOrder[] = [];
+    /** Successful swaps performed for the placed order. */
+    const swapResults: SwapResult[] = [];
+
+    const retryFailedSwap = async (failedSwapQuantity: number) => {
```

Comment:
Unlike the previous version, it looks that you're calling this function for every failed swap, and not for the accumulative failed quantity (of all the failed swaps). Is that on purpose?

Reply:
Yeah it is, just to minimize the total amount of time expected for the order matching routine to complete. If we start retrying a failed swap immediately, it potentially eliminates a wait for other swaps to complete.